src/serac/numerics/functional/functional.hpp

```diff
@@ -280,8 +280,8 @@ class Functional<test(trials...), exec> {
    * and @a spatial_dim template parameter
    * @param[inout] qdata The data for each quadrature point
    */
-  template <int dim, int... args, typename lambda, typename qpt_data_type = Nothing>
-  void AddDomainIntegral(Dimension<dim>, DependsOn<args...>, lambda&& integrand, mfem::Mesh& domain,
+  template <int dim, int... args, typename Integrand, typename qpt_data_type = Nothing>
+  void AddDomainIntegral(Dimension<dim>, DependsOn<args...>, Integrand&& integrand, mfem::Mesh& domain,
```

Comment:
Use `std::forward<Integrand>(integrand)` on line 296 in place of `integrand`. Or just use `Integrand const &` instead of a forwarding reference.